stablehlo/dialect/TypeInference.cpp

```diff
@@ -264,6 +281,76 @@ LogicalResult verifyPairwiseCompatibleShapes(TypeRange values) {
   return success();
 }
 
+LogicalResult verifyAddOp(std::optional<Location> location, Type lhsType,
+                          Type rhsType, Type resultType) {
+  lhsType = getElementTypeOrSelf(lhsType);
+  rhsType = getElementTypeOrSelf(rhsType);
+  resultType = getElementTypeOrSelf(resultType);
+  llvm::SmallVector<Type, 3> typeEntries{lhsType, rhsType, resultType};
+  // add_c1
+  if (noneQuantized<quant::QuantizedType>(typeEntries)) {
```

Comment:
My concern with this change is -- not all users of StableHLO want to think about quantization. It's a subset of total users who use this feature, but they'd have to filter through a very complicated implementation of verifyAdd to understand what their uses need. In that sense, perhaps we can write this function as:

```
if (anyQuantized(lhsType,rhsType,resultType))
  return verifyBinaryOpQuantizationConstraints(...);

return OpTrait::SameOperandsAndResultType<AddOp>::verifyTrait(getOperation());
```

Reply:
Yeh.. agree. I will put the Q Verifier checks under dedicated verifyBinaryOpQuantizationConstraints function.

Reply:
for : `return OpTrait::SameOperandsAndResultType<AddOp>::verifyTrait(getOperation());`

AddOp is  `StableHLO_BinaryElementwiseOp` and shape check is enabled by default. Custom verifier will check  same element type only.
tried  using
`mlir::OpTrait::SameOperandsAndResultElementType<::mlir::stablehlo::AddOp>::verifyTrait(op);` but bazel build failed. There is a circular dependency if I try to add`stablehlo_ops` to `stablehlo_type_inference` target to use `::mlir::stablehlo::AddOp`  at TypeInference.cpp. 

instead trait will do element type check using
`lhsType == resultType ` `lhsType == rhsType`

Reply:
Can we use `SameOperandsAndResultElementType <Operation*>`? Otherwise we can use equality and our own mesage

Reply:
Thank you. yeh.. tried that too before using quality and our own message.